app/src/main/java/org/schabi/newpipe/fragments/list/search/SearchFragment.java

```diff
@@ -395,8 +397,8 @@ public void onSaveInstanceState(@NonNull final Bundle bundle) {
 
     @Override
     public void reloadContent() {
-        if (!TextUtils.isEmpty(searchString)
-                || (searchEditText != null && !TextUtils.isEmpty(searchEditText.getText()))) {
+        if (!TextUtils.isEmpty(searchString) || (searchEditText != null
+                && TextUtils.getTrimmedLength(searchEditText.getText()) > 0)) {
```

Comment:
No need for a new function, use `!isBlank(searchEditText.getText())`, from Android's libraries (also in other places)

Reply:
Should we handle other non-ASCII spaces? Like ```'\u3000'```. 
```String.trim()``` and ```org.schabi.newpipe.extractor.utils.Utils.isBlank()``` do not handle these code points. 
We should use ```String.strip()``` and ```String.isBlank()```, but they do not available in the current version of the Android SDK.

Reply:
There is no need to, in my opinion